lib/puppet/provider/reboot/windows.rb

```diff
@@ -2,8 +2,8 @@
 require 'open3'
 
 Puppet::Type.type(:reboot).provide :windows do
-  confine :operatingsystem => :windows
-  defaultfor :operatingsystem => :windows
+  confine :kernel => :windows
+  defaultfor :kernel => :windows
```

Comment:
On further research, it looks like the built-in Puppet providers confine by operating system (c.f. [exec/windows.rb](https://github.com/puppetlabs/puppet/blob/3.7.3/lib/puppet/provider/exec/windows.rb#L5), [file/windows.rb](https://github.com/puppetlabs/puppet/blob/3.7.3/lib/puppet/provider/file/windows.rb#L4), and [package/windows.rb](https://github.com/puppetlabs/puppet/blob/3.7.3/lib/puppet/provider/package/windows.rb#L23)).  As such, I've changed it in #59.  I can move it to its own PR if you think that'd be preferable.